policy-engine/src/main.rs

```diff
@@ -0,0 +1,223 @@
+    // https://prometheus.io/docs/instrumenting/writing_clientlibs/#process-metrics
+    static ref PROCESS_START_TIME: IntGauge = register_int_gauge!(opts!(
+        "process_start_time_seconds",
+        "Start time of the process since unix epoch in seconds."
+    )).unwrap();
+
+}
+
+fn main() -> Fallible<()> {
+    env_logger::Builder::from_default_env().try_init()?;
+
+    let opts = CliOptions::from_args();
+    trace!("started with CLI options: {:#?}", opts);
+
+    let sys = actix::System::new("dumnati");
+
+    // TODO(lucab): figure out all configuration params.
+    let streams_cfg = maplit::btreeset!["next", "stable", "testing"];
+    let mut scrapers = HashMap::with_capacity(streams_cfg.len());
+    for stream in streams_cfg {
+        let addr = scraper::Scraper::new(stream)?.start();
+        scrapers.insert(stream.to_string(), addr);
+    }
+
+    let node_population = Arc::new(cbloom::Filter::new(10 * 1024 * 1024, 1_000_000));
+    let service_state = AppState {
+        scrapers,
+        population: Arc::clone(&node_population),
+    };
+
+    let start_timestamp = chrono::Utc::now();
+    PROCESS_START_TIME.set(start_timestamp.timestamp());
+
+    // Policy-engine service.
+    let pe_service = service_state.clone();
+    actix_web::HttpServer::new(move || {
+        App::new()
+            .data(pe_service.clone())
+            .route("/v1/graph", web::get().to(pe_serve_graph))
+    })
+    .bind((IpAddr::from(Ipv4Addr::UNSPECIFIED), 8081))?
+    .run();
+
+    // Policy-engine status service.
+    let pe_status = service_state;
+    actix_web::HttpServer::new(move || {
+        App::new()
+            .data(pe_status.clone())
+            .route("/metrics", web::get().to(metrics::serve_metrics))
+    })
+    .bind((IpAddr::from(Ipv4Addr::UNSPECIFIED), 9081))?
+    .run();
+
+    sys.run()?;
+    Ok(())
+}
+
+#[derive(Clone, Debug)]
+pub(crate) struct AppState {
+    scrapers: HashMap<String, Addr<scraper::Scraper>>,
+    population: Arc<cbloom::Filter>,
+}
+
+#[derive(Deserialize)]
+pub struct GraphQuery {
+    basearch: Option<String>,
+    stream: Option<String>,
+    rollout_wariness: Option<String>,
+    node_uuid: Option<String>,
+}
+
+pub(crate) async fn pe_serve_graph(
+    data: actix_web::web::Data<AppState>,
+    actix_web::web::Query(query): actix_web::web::Query<GraphQuery>,
+) -> Result<HttpResponse, Error> {
+    pe_record_metrics(&data, &query);
+
+    let basearch = query
+        .basearch
+        .as_ref()
+        .map(String::from)
+        .unwrap_or_default();
+    let stream = query.stream.as_ref().map(String::from).unwrap_or_default();
+
+    let addr = match data.scrapers.get(&stream) {
+        None => return Ok(HttpResponse::NotFound().finish()),
+        Some(addr) => addr,
+    };
+
+    let wariness = compute_wariness(&query);
+    ROLLOUT_WARINESS.observe(wariness);
+
+    let cached_graph = addr.send(scraper::GetCachedGraph { stream }).await??;
```

Comment:
Self note: this will be changed later on to talk to the graph-builder process running in the same pod.

@zonggen if it helps making the rework cleaner, you can drop all the `Scraper` logic from policy-engine and just insert here a small method which produces an empty graph. We'll first put in place the graph-builder and then connect the policy-engine to it later on (while running the dumnati binary for a bit until everything is in place).

Reply:
Yes I agree to make `fcos-policy-engine` a dummy crate for now and follow up with dedicated PR for that. which makes reviews easier and less error-prone.